### ISSUE TITLE
Can't get it to work. How do I enable write access?

### ISSUE THREAD

**Issue by jedthe3rd** [dreadlocked/Drupalgeddon2#24]
This is what I get when I run it:
[![Screenshot from Gyazo](https://gyazo.com/681b20e39458989cdbb3a11b2ad84b9d/raw)](https://gyazo.com/681b20e39458989cdbb3a11b2ad84b9d)

I am sure it is a problem with how I setup write access but I can't figure out how to fix it.

I am running a local drupal 8.4.5 installation. 
Trying to setup a demo for a school project so any help would be appreciated.

**Comment by g0tmi1k** [comment]
On the target, SSH in then do something like (btw, this is _REALLY_ **BAD IDEA IN PRODUCTION**):

```
chown -R www-data:www-data /var/www/html/
chmod -R 0777 /var/www/html/
```

Not sure exactly how you set it all up.

For me:

```
```
$ ssh root@<REMOVED>
Last login: Wed Apr 18 11:39:52 2018 from <REMOVED>
root@ubuntu140045x64-drupal:~# cd /var/www/html/
root@ubuntu140045x64-drupal:/var/www/html#
root@ubuntu140045x64-drupal:/var/www/html# ls
drupal-7.55  drupal-7.55.tar.gz  drupal-7.57  drupal-7.57.tar.gz  drupal-8.4.5  drupal-8.4.5.tar.gz  index.html
root@ubuntu140045x64-drupal:/var/www/html#
root@ubuntu140045x64-drupal:/var/www/html# chown -R www-data:www-data /var/www/html/
root@ubuntu140045x64-drupal:/var/www/html#
root@ubuntu140045x64-drupal:/var/www/html# chmod -R 0777 /var/www/html/
root@ubuntu140045x64-drupal:/var/www/html#
root@ubuntu140045x64-drupal:/var/www/html# chown -R www-data:www-data /var/www/html/
root@ubuntu140045x64-drupal:/var/www/html# 
root@ubuntu140045x64-drupal:/var/www/html# ls -l
total 19520
drwxrwxrwx 9 www-data www-data     4096 Apr 18 15:08 drupal-7.55
-rwxrwxrwx 1 www-data www-data  3277355 Jun  7  2017 drupal-7.55.tar.gz
drwxrwxrwx 9 www-data www-data     4096 Apr 18 14:28 drupal-7.57
-rwxrwxrwx 1 www-data www-data  3279405 Feb 21 17:45 drupal-7.57.tar.gz
drwxrwxrwx 8 www-data www-data     4096 Apr 18 15:08 drupal-8.4.5
-rwxrwxrwx 1 www-data www-data 13414036 Feb 21 17:42 drupal-8.4.5.tar.gz
-rwxrwxrwx 1 www-data www-data        0 Apr 18 12:58 index.html
root@ubuntu140045x64-drupal:/var/www/html#
```

- - - 

**TL;DR**
- Log in
- Move to the web root
- Set web user (e.g. `www-data`) to own the web folders
- Allow everyone & anyone to read/write/execute all the web root folders

**Comment by dreadlocked** [comment]
You can now use drupalgeddon-not-write-shell.rb which do not require write permissions to work.

**Comment by kostrin** [comment]
I got this working on Ubuntu14.04.5x64 as well but the exploit doesn't work on Ubuntu 16.x on vulnerable Drupal versions. Would you happen to know why?
